pkg/common-controller/framework_test.go

```diff
@@ -1053,6 +1490,64 @@ func newSnapshot(
 	return &snapshot
 }
 
+func newGroupSnapshot(
+	snapshotName, snapshotUID string, selectors map[string]string, targetContentName, snapshotClassName, boundContentName string,
+	readyToUse *bool, creationTime *metav1.Time,
+	err *crdv1.VolumeSnapshotError, nilStatus bool, withAllFinalizers bool, deletionTimestamp *metav1.Time) *crdv1alpha1.VolumeGroupSnapshot {
+	groupSnapshot := crdv1alpha1.VolumeGroupSnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              snapshotName,
+			Namespace:         testNamespace,
+			UID:               types.UID(snapshotUID),
+			ResourceVersion:   "1",
+			SelfLink:          "/apis/groupsnapshot.storage.k8s.io/v1alpha1/namespaces/" + testNamespace + "/volumesnapshots/" + snapshotName,
+			DeletionTimestamp: deletionTimestamp,
+		},
+		Spec: crdv1alpha1.VolumeGroupSnapshotSpec{
+			VolumeGroupSnapshotClassName: nil,
+		},
+	}
+
+	if len(selectors) > 0 {
+		groupSnapshot.Spec.Source.Selector = &metav1.LabelSelector{
+			MatchLabels: selectors,
+		}
+	}
+
+	if !nilStatus {
+		groupSnapshot.Status = &crdv1alpha1.VolumeGroupSnapshotStatus{
+			CreationTime: creationTime,
+			ReadyToUse:   readyToUse,
+			Error:        err,
+		}
+
+		if boundContentName != "" {
+			groupSnapshot.Status.BoundVolumeGroupSnapshotContentName = &boundContentName
+		}
+	}
+
+	if snapshotClassName != "" {
+		groupSnapshot.Spec.VolumeGroupSnapshotClassName = &snapshotClassName
+	}
+
+	if targetContentName != "" {
+		groupSnapshot.Spec.Source.VolumeGroupSnapshotContentName = &targetContentName
+	}
+	if withAllFinalizers {
+		return withGroupSnapshotFinalizers([]*crdv1alpha1.VolumeGroupSnapshot{&groupSnapshot}, utils.VolumeGroupSnapshotContentFinalizer, utils.VolumeGroupSnapshotBoundFinalizer)[0]
+	}
+	return &groupSnapshot
+}
+
+func newGroupSnapshotArray(
+	snapshotName, snapshotUID string, selectors map[string]string, targetContentName, snapshotClassName, boundContentName string,
```

Comment:
same here as well, can we have the inputs to match the group like groupSnapshotName etc

Reply:
Fixed! I changed newGroupSnapshotContent(Array) too